ddtrace/internal/datadog/profiling/dd_wrapper/src/uploader.cpp

```diff
@@ -12,10 +16,33 @@ DdogCancellationTokenDeleter::operator()(ddog_CancellationToken* ptr) const
     }
 }
 
-Datadog::Uploader::Uploader(std::string_view _url, ddog_prof_Exporter* _ddog_exporter)
-  : url{ _url }
+Datadog::Uploader::Uploader(std::string_view _output_filename, ddog_prof_Exporter* _ddog_exporter)
+  : output_filename{ _output_filename }
   , ddog_exporter{ _ddog_exporter }
 {
+    // Increment the upload sequence number every time we build an uploader.
+    // Upoloaders are use-once-and-destroy.
+    upload_seq++;
+}
+
+bool
+Datadog::Uploader::export_to_file(ddog_prof_EncodedProfile* encoded)
+{
+    // Write the profile to a file using the following format for filename:
+    // <output_filename>.<process_id>.<sequence_number>
+    std::ostringstream oss;
+    oss << output_filename << "." << getpid() << "." << upload_seq;
+    std::string filename = oss.str();
+    std::ofstream out(filename.c_str(), std::ios::binary);
+    if (!out.is_open()) {
+        errmsg = "Error opening output file " + filename;
+        std::cerr << errmsg << std::endl;
+        return false;
+    }
+
+    out.write(reinterpret_cast<const char*>(encoded->buffer.ptr), encoded->buffer.len);
```

Comment:
I've seen file writes fail in CI--we should check that the write succeeded (otherwise the error is somewhat opaque)

Reply:
Checks for `fail()`